autoscale/src/main/java/com/sequenceiq/periscope/controller/validation/AlertValidator.java

```diff
@@ -68,6 +68,14 @@ public void validateStopStartEntitlementAndDisableIfNotEntitled(Cluster cluster)
         }
     }
 
+    public void validateImpalaScheduleBasedScalingEntitlement(String clusterCrn, String cloudPlatform, String stackName) {
+        String accountId = Crn.safeFromString(clusterCrn).getAccountId();
+        if (!entitlementValidationService.impalaScheduleBasedScalingEntitlementEnabled(accountId, cloudPlatform)) {
+            throw new BadRequestException(messagesService.getMessage(MessageCode.IMPALA_SCHEDULE_BASED_SCALING_ENTITLEMENT_NOT_ENABLED,
+                    List.of(cloudPlatform, stackName)));
```

Comment:
We should also mention entitlement as part of error message

Reply:
Done